DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -1065,7 +1065,12 @@ public partial class DMParser : Parser<Token> {
                 Consume(TokenType.DM_LeftParenthesis, "Expected '('");
                 BracketWhitespace();
                 DMASTExpression? condition = Expression();
-                if (condition == null) Error("Expected a condition");
+                if (condition == null) {
+                    Error("Expected a condition");
+                }
+                if (condition is DMASTAssign) {
+                    DMCompiler.Emit(WarningCode.AssignmentInConditional, Current().Location, "Assignment in conditional");
```

Comment:
Place the warning at the beginning of the expression rather than after.
```suggestion
                    DMCompiler.Emit(WarningCode.AssignmentInConditional, condition.Location, "Assignment in conditional");
```

Reply:
Resolved